--- .cache/1.16.5/decompiled/net/minecraft/server/dedicated/DedicatedServerProperties.java	2021-06-07 16:20:57.739535009 +1200
+++ src/main/java/net/minecraft/server/dedicated/DedicatedServerProperties.java	2021-06-07 16:30:31.830378106 +1200
@@ -17,7 +17,7 @@
     public final boolean spawnAnimals = this.get("spawn-animals", true);
     public final boolean spawnNpcs = this.get("spawn-npcs", true);
     public final boolean pvp = this.get("pvp", true);
-    public final boolean allowFlight = this.get("allow-flight", false);
+    public final boolean allowFlight = this.get("allow-flight", true);
     public final String resourcePack = this.get("resource-pack", "");
     public final String motd = this.get("motd", "A Minecraft Server");
     public final boolean forceGameMode = this.get("force-gamemode", false);
