--- a/net/minecraft/network/protocol/game/ServerboundPlayerInputPacket.java
+++ b/net/minecraft/network/protocol/game/ServerboundPlayerInputPacket.java
@@ -8,8 +8,8 @@
 public class ServerboundPlayerInputPacket implements Packet<ServerGamePacketListener> {
    private float xxa;
    private float zza;
-   private boolean isJumping;
-   private boolean isShiftKeyDown;
+   public boolean isJumping; // Glucose :: private --> public
+   public boolean isShiftKeyDown; // Glucose :: private --> public
 
    public ServerboundPlayerInputPacket() {
    }
