--- a/net/minecraft/server/dedicated/DedicatedServerProperties.java
+++ b/net/minecraft/server/dedicated/DedicatedServerProperties.java
@@ -17,7 +17,7 @@
    public final boolean spawnAnimals = this.get("spawn-animals", true);
    public final boolean spawnNpcs = this.get("spawn-npcs", true);
    public final boolean pvp = this.get("pvp", true);
-   public final boolean allowFlight = this.get("allow-flight", false);
+   public final boolean allowFlight = this.get("allow-flight", true); // Glucose :: false --> true
    public final String resourcePack = this.get("resource-pack", "");
    public final boolean requireResourcePack = this.get("require-resource-pack", false);
    public final String resourcePackPrompt = this.get("resource-pack-prompt", "");
@@ -129,7 +129,9 @@
    }
 
    // $FF: synthetic method
+   /* // Glucose start
    protected Settings reload(RegistryAccess var1, Properties var2) {
       return this.reload(var1, var2);
    }
+    */ // Glucose end
 }
