--- .cache/1.16.5/decompiled/net/minecraft/server/MinecraftServer.java	2021-06-27 16:20:33.601232028 +1200
+++ src/main/java/net/minecraft/server/MinecraftServer.java	2021-06-27 16:35:29.755792794 +1200
@@ -9,6 +9,7 @@
 import com.mojang.authlib.GameProfileRepository;
 import com.mojang.authlib.minecraft.MinecraftSessionService;
 import com.mojang.datafixers.DataFixer;
+import glucose.Glucose;
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.ByteBufOutputStream;
 import io.netty.buffer.Unpooled;
@@ -923,7 +924,7 @@
    }
 
    public String getServerModName() {
-      return "vanilla";
+      return Glucose.getImplName(); // Glucose :: Set mod name
    }
 
    public CrashReport fillReport(CrashReport var1) {
