--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -53,6 +53,9 @@
 import java.util.stream.Stream;
 import javax.annotation.Nullable;
 import javax.imageio.ImageIO;
+
+import ml.glucosedev.glucose.Glucose;
+import ml.glucosedev.glucose.event.GlucoseEventHandler;
 import net.minecraft.CrashReport;
 import net.minecraft.ReportedException;
 import net.minecraft.SharedConstants;
@@ -175,7 +178,7 @@
    private static final long STATUS_EXPIRE_TIME_NS = 5000000000L;
    private static final int MAX_STATUS_PLAYER_SAMPLE = 12;
    public static final String MAP_RESOURCE_FILE = "resources.zip";
-   public static final File USERID_CACHE_FILE = new File("usercache.json");
+   public static final File USERID_CACHE_FILE = new File("config/usercache.json"); // Glucose :: config dir
    public static final int START_CHUNK_RADIUS = 11;
    private static final int START_TICKING_CHUNK_COUNT = 441;
    private static final int AUTOSAVE_INTERVAL = 6000;
@@ -252,18 +255,20 @@
    private final StructureManager structureManager;
    protected final WorldData worldData;
 
-   public static <S extends MinecraftServer> S spin(Function<Thread, S> var0) {
-      AtomicReference var1 = new AtomicReference();
-      Thread var2 = new Thread(() -> {
-         ((MinecraftServer)var1.get()).runServer();
+   public static <S extends MinecraftServer> S spin(Function<Thread, S> f) {
+      AtomicReference<S> ref = new AtomicReference<S>();
+      Thread thread = new Thread(() -> {
+         (ref.get()).runServer();
       }, "Server thread");
-      var2.setUncaughtExceptionHandler((var0x, var1x) -> {
-         LOGGER.error(var1x);
+
+      thread.setUncaughtExceptionHandler((thread1, throwable) -> {
+         MinecraftServer.LOGGER.error(throwable);
       });
-      MinecraftServer var3 = (MinecraftServer)var0.apply(var2);
-      var1.set(var3);
-      var2.start();
-      return var3;
+      S s0 = (S) f.apply(thread); // Glucose
+
+      ref.set(s0);
+      thread.start();
+      return s0;
    }
 
    public MinecraftServer(Thread var1, RegistryAccess.RegistryHolder var2, LevelStorageSource.LevelStorageAccess var3, WorldData var4, PackRepository var5, Proxy var6, DataFixer var7, ServerResources var8, @Nullable MinecraftSessionService var9, @Nullable GameProfileRepository var10, @Nullable GameProfileCache var11, ChunkProgressListenerFactory var12) {
@@ -313,10 +318,10 @@
    private void readScoreboard(DimensionDataStorage var1) {
       ServerScoreboard var10001 = this.getScoreboard();
       Objects.requireNonNull(var10001);
-      Function var2 = var10001::createData;
+      Function<net.minecraft.nbt.CompoundTag, net.minecraft.world.scores.ScoreboardSaveData> function = var10001::createData; // Glucose
       ServerScoreboard var10002 = this.getScoreboard();
       Objects.requireNonNull(var10002);
-      var1.computeIfAbsent(var2, var10002::createData, "scoreboard");
+//      var1.computeIfAbsent(var2, var10002::createData, "scoreboard"); // Glucose :: Remove line (buggy boi)
    }
 
    protected abstract boolean initServer() throws IOException;
@@ -370,9 +375,9 @@
       long var5 = var3.seed();
       long var7 = BiomeManager.obfuscateSeed(var5);
       ImmutableList var9 = ImmutableList.of(new PhantomSpawner(), new PatrolSpawner(), new CatSpawner(), new VillageSiege(), new WanderingTraderSpawner(var2));
-      MappedRegistry var10 = var3.dimensions();
-      LevelStem var12 = (LevelStem)var10.get(LevelStem.OVERWORLD);
-      Object var11;
+      MappedRegistry<LevelStem> var10 = var3.dimensions();
+      LevelStem var12 = var10.get(LevelStem.OVERWORLD);
+      ChunkGenerator var11;
       DimensionType var13;
       if (var12 == null) {
          var13 = (DimensionType)this.registryHolder.registryOrThrow(Registry.DIMENSION_TYPE_REGISTRY).getOrThrow(DimensionType.OVERWORLD_LOCATION);
@@ -382,7 +387,7 @@
          var11 = var12.generator();
       }
 
-      ServerLevel var14 = new ServerLevel(this, this.executor, this.storageSource, var2, Level.OVERWORLD, var13, var1, (ChunkGenerator)var11, var4, var7, var9, true);
+      ServerLevel var14 = new ServerLevel(this, this.executor, this.storageSource, var2, Level.OVERWORLD, var13, var1, var11, var4, var7, var9, true);
       this.levels.put(Level.OVERWORLD, var14);
       DimensionDataStorage var15 = var14.getDataStorage();
       this.readScoreboard(var15);
@@ -415,15 +420,15 @@
          this.getCustomBossEvents().load(this.worldData.getCustomBossEvents());
       }
 
-      Iterator var17 = var10.entrySet().iterator();
+      Iterator<Entry<ResourceKey<LevelStem>, LevelStem>> var17 = var10.entrySet().iterator();
 
       while(var17.hasNext()) {
-         Entry var27 = (Entry)var17.next();
-         ResourceKey var19 = (ResourceKey)var27.getKey();
+         Entry<ResourceKey<LevelStem>, LevelStem> var27 = var17.next();
+         ResourceKey<LevelStem> var19 = var27.getKey();
          if (var19 != LevelStem.OVERWORLD) {
-            ResourceKey var20 = ResourceKey.create(Registry.DIMENSION_REGISTRY, var19.location());
-            DimensionType var21 = ((LevelStem)var27.getValue()).type();
-            ChunkGenerator var22 = ((LevelStem)var27.getValue()).generator();
+            ResourceKey<Level> var20 = ResourceKey.create(Registry.DIMENSION_REGISTRY, var19.location());
+            DimensionType var21 = (var27.getValue()).type();
+            ChunkGenerator var22 = (var27.getValue()).generator();
             DerivedLevelData var23 = new DerivedLevelData(this.worldData, var2);
             ServerLevel var24 = new ServerLevel(this, this.executor, this.storageSource, var23, var20, var21, var1, var22, var4, var7, ImmutableList.of(), false);
             var16.addListener(new BorderChangeListener.DelegateBorderChangeListener(var24.getWorldBorder()));
@@ -527,7 +532,7 @@
 
       this.nextTickTime = Util.getMillis() + 10L;
       this.waitUntilNextTick();
-      Iterator var5 = this.levels.values().iterator();
+      Iterator<ServerLevel> var5 = this.levels.values().iterator();
 
       while(true) {
          ServerLevel var6;
@@ -587,8 +592,8 @@
    public boolean saveAllChunks(boolean var1, boolean var2, boolean var3) {
       boolean var4 = false;
 
-      for(Iterator var5 = this.getAllLevels().iterator(); var5.hasNext(); var4 = true) {
-         ServerLevel var6 = (ServerLevel)var5.next();
+      for(Iterator<ServerLevel> var5 = this.getAllLevels().iterator(); var5.hasNext(); var4 = true) {
+         ServerLevel var6 = var5.next();
          if (!var1) {
             LOGGER.info("Saving chunks for level '{}'/{}", var6, var6.dimension().location());
          }
@@ -625,6 +630,8 @@
          this.getConnection().stop();
       }
 
+      new GlucoseEventHandler().serverShutdownEvent(); // Glucose
+
       if (this.playerList != null) {
          LOGGER.info("Saving players");
          this.playerList.saveAll();
@@ -632,11 +639,11 @@
       }
 
       LOGGER.info("Saving worlds");
-      Iterator var1 = this.getAllLevels().iterator();
+      Iterator<ServerLevel> var1 = this.getAllLevels().iterator();
 
       ServerLevel var2;
       while(var1.hasNext()) {
-         var2 = (ServerLevel)var1.next();
+         var2 = var1.next();
          if (var2 != null) {
             var2.noSave = false;
          }
@@ -646,7 +653,7 @@
       var1 = this.getAllLevels().iterator();
 
       while(var1.hasNext()) {
-         var2 = (ServerLevel)var1.next();
+         var2 = var1.next();
          if (var2 != null) {
             try {
                var2.close();
@@ -796,10 +803,10 @@
          return true;
       } else {
          if (this.haveTime()) {
-            Iterator var1 = this.getAllLevels().iterator();
+            Iterator<ServerLevel> var1 = this.getAllLevels().iterator();
 
             while(var1.hasNext()) {
-               ServerLevel var2 = (ServerLevel)var1.next();
+               ServerLevel var2 = var1.next();
                if (var2.getChunkSource().pollTask()) {
                   return true;
                }
@@ -810,7 +817,7 @@
       }
    }
 
-   protected void doRunTask(TickTask var1) {
+   public void doRunTask(TickTask var1) {
       this.getProfiler().incrementCounter("runTask");
       super.doRunTask(var1);
    }
@@ -823,11 +830,11 @@
 
       var2.ifPresent((var1x) -> {
          try {
-            BufferedImage var2 = ImageIO.read(var1x);
-            Validate.validState(var2.getWidth() == 64, "Must be 64 pixels wide", new Object[0]);
-            Validate.validState(var2.getHeight() == 64, "Must be 64 pixels high", new Object[0]);
+            BufferedImage var22 = ImageIO.read((File) var1x); // Glucose :: casted to File
+            Validate.validState(var22.getWidth() == 64, "Must be 64 pixels wide", new Object[0]);
+            Validate.validState(var22.getHeight() == 64, "Must be 64 pixels high", new Object[0]);
             ByteArrayOutputStream var3 = new ByteArrayOutputStream();
-            ImageIO.write(var2, "PNG", var3);
+            ImageIO.write(var22, "PNG", var3);
             byte[] var4 = Base64.getEncoder().encode(var3.toByteArray());
             String var10001 = new String(var4, StandardCharsets.UTF_8);
             var1.setFavicon("data:image/png;base64," + var10001);
@@ -901,10 +908,10 @@
       this.profiler.push("commandFunctions");
       this.getFunctions().tick();
       this.profiler.popPush("levels");
-      Iterator var2 = this.getAllLevels().iterator();
+      Iterator<ServerLevel> var2 = this.getAllLevels().iterator();
 
       while(var2.hasNext()) {
-         ServerLevel var3 = (ServerLevel)var2.next();
+         ServerLevel var3 = var2.next();
          this.profiler.push(() -> {
             return var3 + " " + var3.dimension().location();
          });
@@ -1000,7 +1007,7 @@
 
    @DontObfuscate
    public String getServerModName() {
-      return "vanilla";
+      return Glucose.getImplName(); // Glucose :: Set mod name
    }
 
    public SystemReport fillSystemReport(SystemReport var1) {
@@ -1012,22 +1019,22 @@
       }
 
       var1.setDetail("Data Packs", () -> {
-         StringBuilder var1 = new StringBuilder();
+         StringBuilder var10 = new StringBuilder();
          Iterator var2 = this.packRepository.getSelectedPacks().iterator();
 
          while(var2.hasNext()) {
             Pack var3 = (Pack)var2.next();
-            if (var1.length() > 0) {
-               var1.append(", ");
+            if (var10.length() > 0) {
+               var10.append(", ");
             }
 
-            var1.append(var3.getId());
+            var10.append(var3.getId());
             if (!var3.getCompatibility().isCompatible()) {
-               var1.append(" (incompatible)");
+               var10.append(" (incompatible)");
             }
          }
 
-         return var1.toString();
+         return var10.toString();
       });
       if (this.serverId != null) {
          var1.setDetail("Server Id", () -> {
@@ -1093,10 +1100,10 @@
    }
 
    private void updateMobSpawningFlags() {
-      Iterator var1 = this.getAllLevels().iterator();
+      Iterator<ServerLevel> var1 = this.getAllLevels().iterator();
 
       while(var1.hasNext()) {
-         ServerLevel var2 = (ServerLevel)var1.next();
+         ServerLevel var2 = var1.next();
          var2.setSpawnSettings(this.isSpawningMonsters(), this.isSpawningAnimals());
       }
 
@@ -1151,10 +1158,10 @@
       var1.setDynamicData("run_time", (Util.getMillis() - var1.getStartupTime()) / 60L * 1000L);
       var1.setDynamicData("avg_tick_ms", (int)(Mth.average(this.tickTimes) * 1.0E-6D));
       int var2 = 0;
-      Iterator var3 = this.getAllLevels().iterator();
+      Iterator<ServerLevel> var3 = this.getAllLevels().iterator();
 
       while(var3.hasNext()) {
-         ServerLevel var4 = (ServerLevel)var3.next();
+         ServerLevel var4 = var3.next();
          if (var4 != null) {
             var1.setDynamicData("world[" + var2 + "][dimension]", var4.dimension().location());
             var1.setDynamicData("world[" + var2 + "][mode]", this.worldData.getGameType());
@@ -1360,8 +1367,8 @@
    }
 
    public CompletableFuture<Void> reloadResources(Collection<String> var1) {
-      CompletableFuture var2 = CompletableFuture.supplyAsync(() -> {
-         Stream var10000 = var1.stream();
+      CompletableFuture<Void> var2 = CompletableFuture.supplyAsync(() -> {
+         Stream<String> var10000 = var1.stream(); // Glucose -- Ty Loom :P
          PackRepository var10001 = this.packRepository;
          Objects.requireNonNull(var10001);
          return (ImmutableList)var10000.map(var10001::getPack).filter(Objects::nonNull).map(Pack::open).collect(ImmutableList.toImmutableList());
@@ -1369,10 +1376,10 @@
          return ServerResources.loadResources(var1x, this.registryHolder, this.isDedicatedServer() ? Commands.CommandSelection.DEDICATED : Commands.CommandSelection.INTEGRATED, this.getFunctionCompilationLevel(), this.executor, this);
       }).thenAcceptAsync((var2x) -> {
          this.resources.close();
-         this.resources = var2x;
+         this.resources = (ServerResources) var2x; // Glucose :: Casted to ServerResources
          this.packRepository.setSelected(var1);
          this.worldData.setDataPackConfig(getSelectedPacks(this.packRepository));
-         var2x.updateGlobals();
+         ((ServerResources) var2x).updateGlobals(); // Glucose :: Casted to ServerResources
          this.getPlayerList().saveAll();
          this.getPlayerList().reloadResources();
          this.functionManager.replaceLibrary(this.resources.getFunctionLibrary());
@@ -1426,8 +1433,8 @@
    }
 
    private static DataPackConfig getSelectedPacks(PackRepository var0) {
-      Collection var1 = var0.getSelectedIds();
-      ImmutableList var2 = ImmutableList.copyOf(var1);
+      Collection<String> var1 = var0.getSelectedIds();
+      ImmutableList<String> var2 = ImmutableList.copyOf(var1);
       List var3 = (List)var0.getAvailableIds().stream().filter((var1x) -> {
          return !var1.contains(var1x);
       }).collect(ImmutableList.toImmutableList());
@@ -1813,6 +1820,7 @@
    }
 
    // $FF: synthetic method
+   /* // Glucose Start :: Compile error
    public void doRunTask(Runnable var1) {
       this.doRunTask((TickTask)var1);
    }
@@ -1826,6 +1834,7 @@
    public Runnable wrapRunnable(Runnable var1) {
       return this.wrapRunnable(var1);
    }
+    */ // Glucose end
 
    static {
       DEMO_SETTINGS = new LevelSettings("Demo World", GameType.SURVIVAL, false, Difficulty.NORMAL, false, new GameRules(), DataPackConfig.DEFAULT);
